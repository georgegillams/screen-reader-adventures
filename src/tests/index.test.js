import React from 'react';
// import { act } from 'react-dom/test-utils';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import configureStore from 'client-utils/common/redux/configure-store';

import IndexPage from '../pages/index';

describe('<HomePage />', () => {
  let store;

  beforeAll(() => {
    store = configureStore({});
  });

  it('should render correctly', () => {
    let props;

    // If the page defines `getInitialProps` we should call them here
    // act(() => {
    //   props = IndexPage.getInitialProps();
    // });

    const { container } = render(
      <Provider store={store}>
        <IndexPage {...props} />
      </Provider>
    );

    expect(container).toMatchSnapshot();
  });
});
