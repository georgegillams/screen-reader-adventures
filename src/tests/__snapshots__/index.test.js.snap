// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HomePage /> should render correctly 1`] = `
<div>
  <main
    class="page-container__container common-layout__grow"
    id="main"
  >
    <div
      class="page-container__container page-container__container--regular page-container__container--center"
    >
      <h1
        class="page-title__heading"
      >
        Home
      </h1>
      <div
        class="subsection__outer"
      >
        <span
          class="paragraph"
        >
          This is just a boilerplate page, but you can put any content you want in here.
          <br />
          This page is pretty boring, but the 
          <a
            class="text-link text-link--text-link text-link--no-padding text-link--inline"
            href="/sitemap"
          >
            site map
          </a>
           might be more inspiring.
        </span>
      </div>
    </div>
  </main>
  <footer
    class="footer__container"
    id="footer"
  >
    <div
      class="logo__container footer__logo logo__container--centred logo__container--no-padding"
    >
      <a
        class="logo__a"
        href="/"
      >
        <h1
          aria-label="Home page"
          class="logo__heading logo__heading--no-padding"
          role="button"
        >
          LOGO
        </h1>
      </a>
    </div>
    <div
      class="footer__tech"
    >
      <span
        class="paragraph tech-specs__container"
        theme="allWhite"
      >
        Built in
        <a
          href="https://reactjs.org/"
          rel="noopener noreferrer"
          target="_blank"
        >
          <img
            alt="React"
            class="tech-specs__icon"
            height="5"
            src="IMAGE_MOCK"
            width="5"
          />
        </a>
        and
        <a
          href="https://redux.js.org/"
          rel="noopener noreferrer"
          target="_blank"
        >
          <img
            alt="Redux"
            class="tech-specs__icon"
            height="5"
            src="IMAGE_MOCK"
            width="5"
          />
        </a>
        Hosted on
        <a
          href="https://aws.amazon.com/"
          rel="noopener noreferrer"
          target="_blank"
        >
          <img
            alt="Amazon Web Services"
            class="tech-specs__icon"
            height="5"
            src="IMAGE_MOCK"
            style="margin-top: .8rem; max-width: 2.9rem;"
            width="8.28"
          />
        </a>
      </span>
    </div>
    <div
      class="footer-nav__container footer__nav"
    >
      <a
        class="text-link text-link--text-link text-link--no-padding text-link--text-link--all-white text-link--inline footer-nav__link"
        href="/sitemap"
      >
        Site map
      </a>
      <a
        class="text-link text-link--text-link text-link--no-padding text-link--text-link--all-white text-link--inline footer-nav__link"
        href="/status"
      >
        Status
      </a>
      <a
        class="text-link text-link--text-link text-link--no-padding text-link--text-link--all-white text-link--inline footer-nav__link"
        href="/privacy-policy"
      >
        Privacy
      </a>
    </div>
  </footer>
</div>
`;
